src/overlays/actors/ovl_En_Sekihi/z_en_sekihi.c

```diff
@@ -32,20 +36,147 @@ const ActorInit En_Sekihi_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+static s16 sObjectIds[] = { OBJECT_SEKIHIL, OBJECT_SEKIHIG, OBJECT_SEKIHIN, OBJECT_SEKIHIZ, OBJECT_ZOG };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sekihi/EnSekihi_Init.s")
+static s32 sOpaDLists[] = { &gGraveTriforceSymbolDL, &gGraveGoronSymbolDL, &gGraveKokiriSymbolDL, &gTombSongOfSoaringDL,
+                            &gMikauGraveDL };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sekihi/EnSekihi_Destroy.s")
+static s32 sXluDLists[] = { &gSunSongTriforceSymbolDL, &gSunSongGoronSymbolDL, &gSunSongKokiriSymbolDL,
+                            &gSekihizEmptyDL, &gMikauGraveDirtDL };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sekihi/func_80A44DE8.s")
+static u16 sTextIds[] = { 0, 0, 0, 0, 0x1018 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sekihi/func_80A44F40.s")
+void EnSekihi_Init(Actor* thisx, PlayState* play) {
+    EnSekihi* this = THIS;
+    s32 params = ENSIKIHI_GET_F(thisx);
+    s32 objectIndex;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sekihi/func_80A450B0.s")
+    if ((params < 0 || params >= 5) || sOpaDLists[params] == 0) {
```

Comment:
```suggestion
    if ((params < 0 || params >= 5) || sOpaDLists[params] == NULL) {
```
Also DLists are pointers so should check against NULL